app/src/main/java/com/matxowy/flashcardsapp/presentation/addflashcardsscreen/view/AddFlashcardsFragment.kt

```diff
@@ -1,18 +1,126 @@
 package com.matxowy.flashcardsapp.presentation.addflashcardsscreen.view
 
+import android.content.Context
 import android.os.Bundle
 import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import androidx.core.view.isVisible
+import androidx.core.widget.addTextChangedListener
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.viewModels
+import com.google.android.material.snackbar.Snackbar
 import com.matxowy.flashcardsapp.R
+import com.matxowy.flashcardsapp.app.utils.extensions.observeWithLifecycle
+import com.matxowy.flashcardsapp.app.utils.hideKeyboard
+import com.matxowy.flashcardsapp.data.db.entity.Category
+import com.matxowy.flashcardsapp.databinding.AddFlashcardsFragmentBinding
 import com.matxowy.flashcardsapp.presentation.addflashcardsscreen.viewmodel.AddFlashcardsViewModel
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class AddFlashcardsFragment : Fragment(R.layout.add_flashcards_fragment) {
+
     private val viewModel: AddFlashcardsViewModel by viewModels()
+    private var _binding: AddFlashcardsFragmentBinding? = null
+    private val binding get() = _binding!!
+
+    private var flashcardTextFront = DEFAULT_EMPTY_STRING
+    private var flashcardTextBack = DEFAULT_EMPTY_STRING
+    private var isCategorySelected = false
```

Comment:
Fragment nie trzyma stanu w polach. Jeśli potrzebujesz takiej flagi to powinna być w viewModelu. Front i back tekst też nie powinien być tu trzymany.

Reply:
Poprawione 😄